services/kvstoremanager/redis/redis.go

```diff
@@ -169,10 +174,46 @@ func (m *RedisManager) HSet(hash, key string, value interface{}) (err error) {
 	return err
 }
 
-func (m *RedisManager) ExtractJSONSetArgs(jsonData json.RawMessage) ([]string, error) {
-	key := gjson.GetBytes(jsonData, "message.key").String()
-	path := gjson.GetBytes(jsonData, "message.path").String()
-	jsonVal := gjson.GetBytes(jsonData, "message.value")
+type NonRootInsertReturn struct {
+	SetArgsPath string
+	MergeTo     string
+	MergeFrom   interface{}
+}
+
+func (m *RedisManager) setArgsForMergeStrategy(inputArgs setArguments) ([]string, error) {
+	isRootInsert := inputArgs.path == ""
+	args := []string{inputArgs.key, "$", inputArgs.jsonVal.String()}
+	redisValueForKey, err := m.GetClient().JSONGet(context.Background(), inputArgs.key).Result()
+	if err != nil {
+		return nil, err
+	}
+
+	valueToBeInserted := redisValueForKey // value to which the transformed value should be merged which will be inserted into Redis
+	if redisValueForKey == "" {
+		valueToBeInserted = "{}"
+	}
+
+	var mergeFrom string = inputArgs.jsonVal.String() // transformed value
+	if !isRootInsert {
+		mergedTo, setErr := sjson.Set("{}", inputArgs.path, inputArgs.jsonVal.Value())
+		if setErr != nil {
+			return nil, fmt.Errorf("problem while setting value into path: %w", setErr)
+		}
+		mergeFrom = mergedTo
+	}
+	// merge jsons
+	mergedValueToBeInserted, mergedErr := jsonpatch.MergeMergePatches([]byte(valueToBeInserted), []byte(mergeFrom))
+	if mergedErr != nil {
+		return nil, fmt.Errorf("merging process failed: %w", mergedErr)
+	}
+	args[2] = string(mergedValueToBeInserted)
+	return args, nil
+}
+
+func (m *RedisManager) setArgsForReplaceStrategy(inputArgs setArguments) ([]string, error) {
```

Comment:
should we have return arguments at struct rather than string array for better maintainability

Reply:
Addressed